server/status_test.go

```diff
@@ -118,9 +130,96 @@ func TestStatusJson(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+			t.Log(string(body))
 			if matches, err := regexp.Match(spec.expected, body); !matches || err != nil {
 				t.Errorf("expected match: %t; err nil: %s", matches, err)
 			}
 		}
 	}
 }
+
+// TestStatusGossipJson ensures that the output requrest for the full gossip
+// info contains the required fields.
+func TestStatusGossipJson(t *testing.T) {
+	s := StartTestServer(t)
+	defer s.Stop()
+
+	type prefixedInfo struct {
+		Key string                 `json:"Key"`
+		Val []storage.PrefixConfig `json:"Val"`
+	}
+
+	type rangeDescriptorInfo struct {
+		Key string                `json:"Key"`
+		Val proto.RangeDescriptor `json:"Val"`
+	}
+
+	type keyValueStringPair struct {
+		Key string `json:"Key"`
+		Val string `json:"Val"`
+	}
+
+	type keyValuePair struct {
+		Key string                 `json:"Key"`
+		Val map[string]interface{} `json:"Val"`
+	}
+
+	type infos struct {
```

Comment:
That was what I had originally actually, but I'm planning on adding multiple nodes and stores to the test server soon and actually inspecting the innards of the returned values, so this will be more useful shortly.